common/render.go

```diff
@@ -136,44 +136,58 @@ func addCameraSystemOnce(w *ecs.World) {
 }
 
 func (rs *RenderSystem) Add(basic *ecs.BasicEntity, render *RenderComponent, space *SpaceComponent) {
-	// Setting default shader
-	if render.shader == nil {
-		switch render.Drawable.(type) {
-		case Triangle:
-			render.shader = LegacyShader
-		case Circle:
-			render.shader = LegacyShader
-		case Rectangle:
-			render.shader = LegacyShader
-		case ComplexTriangles:
-			render.shader = LegacyShader
-		case Text:
-			render.shader = TextShader
-		default:
-			render.shader = DefaultShader
+	if rs.EntityExists(basic) == -1 {
```

Comment:
[Idiomatic Go](https://talks.golang.org/2013/bestpractices.slide#2) would be to put the exceptional case on the top (with a return), and then the 'regular logic' behind it. 

```go
if rs.EntityExists(basic) >= 0 {
    return
}

// Setting default shader
// ...
```